src/plugins/intel_cpu/src/emitters/snippets/x64/jit_brgemm_emitter.cpp

```diff
@@ -250,65 +207,77 @@ void jit_brgemm_emitter::emit_brgemm_kernel_call(const brgemm_kernel_t *brg_kern
+
+    auto reserved_stack_size = sizeof(jit_brgemm_call_args);
+#ifdef _WIN32
+    // Before function call we should also allocate stack area for ABI parameters (shadow space)
+    reserved_stack_size += 3 * gpr_size;
+#endif
+
+    // Reserve memory on the stack
+    h->sub(h->rsp, reserved_stack_size);
+
     const auto data_ptr_reg = [&](Xmm xmm, Xbyak::Reg64 reg, size_t bytes_offset) {
         h->uni_vmovq(reg, xmm);
         if (bytes_offset) h->add(reg, bytes_offset);
     };
-    h->mov(abi_param1, reinterpret_cast<uintptr_t>(brg_kernel));
-    data_ptr_reg(Xmm(0), abi_param2, in0_kernel_offset);
-    data_ptr_reg(Xmm(1), abi_param3, in1_kernel_offset);
-    data_ptr_reg(Xmm(2), abi_param4, out0_kernel_offset);
 
-#ifdef _WIN32
-    // Before function call we should allocate stack area for
-    //  - register parameters - ABI parameters (shadow space)
-    //  - stack parameters - remaining parameters
-    const size_t num_args_passed_on_stack = 6;  // count of function brgemm_kernel_overload() parameters
-    size_t abi_param_count = sizeof(abi_param_regs) / sizeof(abi_param_regs[0]);
-    h->sub(h->rsp, num_args_passed_on_stack * gpr_size);
-
-    // Push the remaining parameters on the stack
-    if (m_with_scratch) {
-        h->uni_vmovq(h->qword[h->rsp + (abi_param_count + 0) * gpr_size], Xmm(3));
-        if (in2_kernel_offset) h->add(h->qword[h->rsp + (abi_param_count + 0) * gpr_size], in2_kernel_offset);
-    } else {
-        h->mov(h->qword[h->rsp + (abi_param_count + 0) * gpr_size], reinterpret_cast<uintptr_t>(nullptr));
-    }
-    h->mov(abi_not_param1, static_cast<int>(m_with_comp));
-    h->mov(h->qword[h->rsp + (abi_param_count + 1) * gpr_size], abi_not_param1);
-#else
+    data_ptr_reg(Xmm(0), h->r8, in0_kernel_offset);
+    h->mov(h->qword[h->rsp + GET_OFF_BRGEMM_ARGS(A)], h->r8);
+
+    data_ptr_reg(Xmm(1), h->r9, in1_kernel_offset);
+    h->mov(h->qword[h->rsp + GET_OFF_BRGEMM_ARGS(B)], h->r9);
+
+    data_ptr_reg(Xmm(2), h->r10, out0_kernel_offset);
+    h->mov(h->qword[h->rsp + GET_OFF_BRGEMM_ARGS(C)], h->r10);
+
     if (m_with_scratch) {
-        data_ptr_reg(Xmm(3), abi_param5, in2_kernel_offset);
+        data_ptr_reg(Xmm(3), h->r11, in2_kernel_offset);
+        h->mov(h->qword[h->rsp + GET_OFF_BRGEMM_ARGS(scratch)], h->r11);
     } else {
-        h->mov(abi_param5, reinterpret_cast<uintptr_t>(nullptr));
+        h->mov(h->qword[h->rsp + GET_OFF_BRGEMM_ARGS(scratch)], reinterpret_cast<uintptr_t>(nullptr));
     }
-    h->mov(abi_param6, static_cast<int>(m_with_comp));
-#endif
+
+    // abi_param1 always contains jit_snippets_call_args which has amx tile config for each thread
+    h->mov(h->r12, h->ptr[abi_param1 + GET_OFF(amx_tile_config)]);
+    h->mov(h->qword[h->rsp + GET_OFF_BRGEMM_ARGS(amx_tile_config)], h->r12);
+
+    h->mov(abi_param1, reinterpret_cast<uintptr_t>(brg_kernel));
+    h->mov(abi_param2, reinterpret_cast<uintptr_t>(&m_ctx));
+    h->lea(abi_param3, h->ptr[h->rsp]);
```

Comment:
Interesting 🤔 
Why can't we just move the content of the `rsp` reg?  `h->mov(abi_param3, h->rsp)`

Reply:
We can actually 😄 Modified, thanks